### PR TITLE
Edit Profile

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello reference

* [As a user I should be able to edit my profile so I can update my personal information or fill incomplete fields](https://trello.com/c/8PzL913m)

#### Descriptions

* Update password
* Delete user account

Screen Shot 2020-03-27 at 9 47 57 AM

### CODE REVIEW

File: src/pages/ProfilePage/index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+
+import smilies from 'assets/smilies.svg';
+import defaultProfile from 'assets/default_profile.svg';
+
+import { getProfile, updatePassword, deleteAccount } from 'redux/actions/userActions';
+import FormInput from 'components/common/formInput/FormInput';
+import useForm from 'hooks/useForm';
+import Button from 'components/common/button/Button';
+import withMap from 'components/HOCs/withMap';
```
Comment (lucia-santos) [inline src/pages/ProfilePage/index.js]: Just to have the code more organized it is good to have all imports related to each other together. For example, as you did when importing the assets you can do it with the components.

File: src/redux/reducers/userReducer.js [hunk 2]
```diff
@@ -49,6 +53,20 @@ export default createReducer(intialState, {
...
+    error:
+      (payload && payload.full_messages && payload.full_messages[0]) ||
+      'An error occured. Please try again.'
+  }),
+  [deleteAccountSuccess]: () => ({
+    intialState
+  }),
+  [deleteAccountError]: (state, { payload }) => ({
+    ...state,
+    error: payload?.full_messages?.[0] || 'An error occurred. Please try again.'
```
Comment (lucia-santos) [inline src/redux/reducers/userReducer.js:83]: I see that you are using this condition `payload?.full_messages?.[0] || 'An error occurred. Please try again.'` more than once, so maybe you can create a function for this with the payload as a param.

**Review by lucia-santos — APPROVED** [review]
Good job!

File: src/redux/reducers/userReducer.js (hunk 2 above)
Comment (afmicc) [reply src/redux/reducers/userReducer.js:83]: Great idea! I'll add these changes at a new PR

File: src/redux/reducers/userReducer.js [hunk 3]
```diff
@@ -49,6 +53,20 @@ export default createReducer(intialState, {
   [logoutError]: (state, { payload }) => ({
     ...state,
     error:
-      payload?.full_messages?.[0] || 'An error occurred. Please try again.' 
+      payload?.full_messages?.[0] || 'An error occurred. Please try again.'
+  }),
+  [updatePasswordSuccess]: (state, { payload }) => ({ value: payload }),
```
Comment (peter7z) [inline src/redux/reducers/userReducer.js]: You are removing `auth` from the state auth here, is that on purpose?

File: src/pages/ProfilePage/index.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+const ProfilePage = () => {
+  const user = useSelector(({ user: { value } }) => value);
+  const { data, validation, validForm, handleFieldChange } = useForm(config);
+
+  const error = useSelector(({ user: { error } }) => error);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getProfile(data));
```
Comment (peter7z) [inline src/pages/ProfilePage/index.js]: This doesn't seem right. `data` is an empty user object that comes from `useForm`. And if I'm not mistaken you need a user ID. Wha am I missing here?

File: src/redux/reducers/userReducer.js (hunk 3 above)
Comment (afmicc) [reply src/redux/reducers/userReducer.js]: No, it's a mistake.

**Review by peter7z — APPROVED** [review]

**Pull request merged**